AURN_Data_Download/AURN_download.py

```diff
@@ -626,53 +688,65 @@ def test_preprocess_code(df_in,pt,spc_zero_process = ['O3','NO2','NOXasNO2'],min
     # If a single year is passed then convert to a list with a single value
     if type(years) is int:
         years = [years]
+    print('Years:', years)
     current_year = datetime.datetime.now()
 
 
     base_path = Path("AURN_data_download")
-    if (base_path.is_dir() is False): 
-        base_path.mkdir() 
-    # change this if we need more organised data storage later  
+    if (base_path.is_dir() is False):
+        base_path.mkdir()
+    # change this if we need more organised data storage later
     data_path = base_path
 
-    # get list of sites to process
-    site_list = metadata['AURN_metadata']['site_id'].unique()
-    
+    # get list of sites to processextract_site_data
+    if not site_list:
+        site_list = metadata['AURN_metadata']['site_id'].unique()
+    else:
+        # Todo - Test that sites provided are correct
+        pass
+    print('Site list', site_list)
+    #sys.exit(0)
+
     # create the station location dataset
     stations = metadata['AURN_metadata'][['site_id','latitude', 'longitude','site_name']].drop_duplicates()
     stations = stations.rename(columns={"site_id":"SiteID","latitude":"Latitude","longitude":"Longitude"})
     stations = stations.set_index('SiteID')
-    
-    
+
+
     # create a dataframe with the hourly dataset for all stations
-    #daily_dataframe = extract_site_data(site_list,metadata,years,data_path,save_to_csv)
-    hourly_dataframe = extract_site_data(site_list,metadata,years,data_path,save_to_csv)
-    hourly_dataframe= hourly_dataframe.rename(columns={'siteID':'SiteID'})
-    
+    hourly_dataframe = extract_site_data(site_list, metadata, years, data_path, save_to_csv)
+    hourly_dataframe = hourly_dataframe.rename(columns={'siteID':'SiteID'})
+
     # apply some filtering of negative and zero values
     spc_list = ['O3','PM10','PM2.5','NO2','NOXasNO2','SO2']
     for spc in spc_list:
-        print('{} has {} positive values'.format(spc,len(hourly_dataframe[spc][hourly_dataframe[spc]>0.0])))
-        print('{} has {} NaNs'.format(spc,len(hourly_dataframe[spc][hourly_dataframe[spc].isna()])))
-        print('{} has {} negative or zero values that will be replaced with NaNs'.format(spc,len(hourly_dataframe[spc][hourly_dataframe[spc]<=0.0])))
-        hourly_dataframe[spc][hourly_dataframe[spc]<=0.0]   = np.nan
+        print('filtering {}:'.format(spc))
+        try:
+            print('\t{} has {} positive values'.format(spc,len(hourly_dataframe.loc[hourly_dataframe[spc]>0.0])))
+            print('\t{} has {} NaNs'.format(spc,len(hourly_dataframe.loc[hourly_dataframe[spc].isna()])))
+            print('\t{} has {} negative or zero values that will be replaced with NaNs'.format(spc,len(hourly_dataframe.loc[hourly_dataframe[spc]<=0.0])))
+            hourly_dataframe.loc[hourly_dataframe[spc]<=0.0, 'spc'] = np.nan
```

Comment:
is this the location of our error - spc shouldn't be in a string?

Reply:
Oh yes!!